proposals/lip-0043.md

```diff
@@ -197,85 +164,180 @@ sidechainRegParams = {
+```python
+def verify(trs: Transaction) -> None:
+    # The name property has to contain only characters from the set [a-z0-9!@$&_.].
+    if not re.match(r"^[a-z0-9!@$&_.]+$", trs.params.name):
+        raise Exception("Invalid name property. It should contain only characters from the set [a-z0-9!@$&_.].")
+    # The name property has to be unique with respect to the set of already registered sidechain names.
+    if there exists an entry in the registered names substore with store key equal to trs.params.name.encode("utf-8"):
+        raise Exception("Name already registered.")
+
+    # Chain ID has to be unique with respect to the set of already registered sidechains.
+    if there exists an entry in the chain data substore with store key equal to trs.params.chainID:
+        raise Exception("Chain ID already registered.")
+    
+    # initValidators must have at least 1 element and at most MAX_NUM_VALIDATORS elements.
+    if len(trs.params.initValidators) < 1 or len(trs.params.initValidators) > MAX_NUM_VALIDATORS:
+        raise Exception(f"Invalid initValidators array. It must have at least 1 element and at most {MAX_NUM_VALIDATORS} elements.")
+    
+    validatorKeys = [validator["blsKey"] for validator in trs.params.initValidators]
+    # All validator keys must be distinct.
+    if len(validatorKeys) != len(set(validatorKeys)):
+        raise Exception("Duplicate BLS keys.")
+    
+    # Validator keys must be in lexicographic order.
+    if not all(validatorKeys[i] < validatorKeys[i + 1] for i in range(len(validatorKeys) - 1)):
+        raise Exception("Validator keys are not in lexicographic order.")
+    
+    totalWeight = 0
+    for validator in trs.params.initValidators:
+        # The bftWeight property of each element is a positive integer.
+        if validator["bftWeight"] == 0:
+            raise Exception("Invalid bftWeight property.")
+        totalWeight += validator["bftWeight"]
+    # Total BFT weight has to be less than or equal to MAX_UINT64.
+    if totalWeight > MAX_UINT64:
+        raise Exception("Total BFT weight exceeds maximum value.")
+    
+    # The range of valid values of the certificateThreshold property is given by the total sum of the validators weights:
+    # Minimum value: floor(1/3 * total BFT weight) + 1.
+    # Maximum value = total BFT weight.
+    if trs.params.certificateThreshold < totalWeight//3 + 1:
+        raise Exception("Certificate threshold is too small.")
+    if trs.params.certificateThreshold > totalWeight:
+        raise Exception("Certificate threshold is too large.")
+
+    # sidechainRegistrationFee must equal REGISTRATION_FEE.
+    if trs.params.sidechainRegistrationFee != REGISTRATION_FEE:
+        raise Exception("Invalid extra command fee.")
+    # Sender must have enough balance to pay for extra command fee.
+    senderAddress = sha256(trs.senderPublicKey)[:ADDRESS_LENGTH]
+    if Token.getAvailableBalance(senderAddress, TOKEN_ID_LSK_MAINCHAIN) < REGISTRATION_FEE:
```

Comment:
we cannot check that sum because the fee is responsibility of the fee module. Indeed, we could have edge cases that pass the verification but would still fail in the execution (so it's not a security issue)